api/src/main/java/eu/darkbot/api/managers/StarSystemAPI.java

```diff
@@ -39,13 +40,46 @@ public interface StarSystemAPI extends API.Singleton {
      */
     Collection<? extends GameMap> getMaps();
 
+    /**
+     * Find {@link GameMap} by given {@code mapId} otherwise will create a new one with given mapId.
+     *
+     * @param mapId id to find
+     * @return {@link GameMap} with given {@code mapId}
+     */
+    GameMap getOrCreateMap(int mapId);
+
+    /**
+     * Find {@link GameMap} by given {@code mapName} otherwise will create a new one with given mapName.
+     *
+     * @param mapName name to find
+     * @return {@link GameMap} with given {@code mapId}
+     */
+    GameMap getOrCreateMap(String mapName);
```

Comment:
I don't think this makes sense, we cannot create a map if we do not know the ID. When searching by id it's easy to autogenerate a name (eg: "Unknown map [id]") but it doesn't make sense the other way around.